task1/task1_2.js

```diff
@@ -0,0 +1,29 @@
+import { createReadStream, createWriteStream } from 'fs';
+import { Transform } from 'stream';
+import { createInterface } from 'readline';
+import csvtojson from 'csvtojson';
+
+const csvFolderPath = './task1/csv';
+const csvFileName = 'books.csv';
+const csvFilePath = `${csvFolderPath}/${csvFileName}`;
+const nowDate = new Date();
+const dateStr = nowDate.toISOString().replace(/:/g, '.');
+const txtFileName = `books_${dateStr}.txt`;
+const txtFilePath = `${csvFolderPath}/${txtFileName}`;
+
+const csvFile = createReadStream(csvFilePath, { encoding: 'utf-8' });
+const readlineInterface = createInterface({ input: csvFile });
+const csvByLine = new Transform({
+  transform: (buf, encoding, callback) => {
+    callback(null, buf);
+  }
+});
+const converter = csvtojson({ delimiter: ';' }, { encoding: 'utf-8' });
+const writeToTxt = createWriteStream(txtFilePath, { encoding: 'utf-8' });
+
+csvFile.on('error', (err) => console.error('read error\n', err.message));
+converter.on('error', (err) => console.error('cvstojson error\n', err.message));
+writeToTxt.on('error', (err) => console.error('write error\n', err.message));
```

Comment:
Also, you can use `try {} catch (err) {}` with one `console.log()` for catch any error because `err` include all necessary information

Reply:
But, I am not sure for 100%. I think we have to ask it

Reply:
I will create one callback for all errors, thank you

Reply:
I didn't managed how to do this